presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -496,6 +506,141 @@ public CounterStat getHighMemorySplitSource()
         return concat(partitionBatches);
     }
 
+    private static PartitionBatchInfo getPartitionBatchInfo(
+            ConnectorSession session,
+            SemiTransactionalHiveMetastore metastore,
+            SchemaTableName tableName,
+            List<HivePartition> partitionBatch,
+            Map<String, HiveColumnHandle> predicateColumns,
+            Optional<Map<Subfield, Domain>> domains)
+    {
+        if (isPartitionStatsBasedOptimizationEnabled(session) && domains.isPresent()) {
+            Map<String, Optional<PartitionWithStatistics>> batch = metastore.getPartitionsWithStatisticsByNames(
+                    tableName.getSchemaName(),
+                    tableName.getTableName(),
+                    Lists.transform(partitionBatch, HivePartition::getPartitionId));
```

Comment:
We can reuse our existing infra? https://github.com/highker/presto/commit/1a64f5b79225957e3faf93cefb8bc6e1e58d8bd3